rosidl_generator_py/cmake/rosidl_generator_py_generate_interfaces.cmake

```diff
@@ -184,7 +184,7 @@ set_property(
 
 macro(set_properties _build_type)
   set_target_properties(${_target_name} PROPERTIES
-    COMPILE_FLAGS "${_extension_compile_flags}"
+    COMPILE_FLAGS ${_extension_compile_flags}
```

Comment:
Currently the variable contains a string. Have you tried to convert it into a list and then use `COMPILE_OPTIONS`?